srcStatic/OP2Memory.cpp

```diff
@@ -22,7 +22,7 @@ void SetLoadOffset()
 template <typename Function>
 bool Op2MemEdit(void* destBaseAddr, std::size_t size, Function memoryEditFunction)
 {
-	void* destAddr = (void*)(loadOffset + (int)destBaseAddr);
+	void* destAddr = reinterpret_cast<void*>(reinterpret_cast<std::size_t>(destBaseAddr) + loadOffset);
```

Comment:
Use `static_cast` when there is an conversion operator between the types (`operator typeName(...)`). This works for converting between the various primitive types such as `char`, `short`, `int`, the `unsigned` versions of them, and `float`, `double`, `long double`. You may also use it on classes that have a defined type conversion operator. The `static_cast` effectively calls the conversion operator. There are some additional details, such as self-type conversions, and conversions involving unions and structs, though most uses seem to involve a conversion operator on a primitive type, or on a class.

Use [`reinterpret_cast`](https://en.cppreference.com/w/cpp/language/reinterpret_cast) when you're converting between pointers to unrelated types, such as a `void*` and an `int*`, or between a `void*` and a `std::size_t`. The `reinterpret_cast` does not generate any assembly instructions, it merely defeats the static type checking done at compile time. You're basically telling the compiler to treat access to that memory as if it was a different type (and possibly a different size).

Use `dynamic_cast` when you are converting between pointers to objects in the same hierarchy. You can always cast up to a base class easily enough, but you need a `dynamic_cast` to down cast from a base pointer to a derived pointer.

And `const_cast` is easier, as it can cast away `const`. It also doesn't generate any assembly instructions. It merely defeats the types checks done at compile time.

Reply:
I just noticed that I'd switched the inner data type from `void*` to `std::size_t` here. Hence your question makes quite a bit of sense, and was maybe hinting at that. The code is no longer converting between pointer types, even though it's still converting between pointer values. I'm uncertain which cast I should use when converting between pointer types and `std::size_t`.

It's probably still fine. Maybe we'll revisit later if experience teaches us something new.